walkthroughs/eks-app-mesh-cross-account-acm/mesh/apply_tls_on_yelb_nodes.sh

```diff
@@ -0,0 +1,75 @@
+
+aws --profile backend ram accept-resource-share-invitation --resource-share-invitation-arn $CA_RESOURCE_INVITE_ARN
+
+
+#UpdateVirtualNodes
+export backend_certificate_arn=$(aws acm request-certificate --domain-name "*.example.com" --certificate-authority-arn $certificateAuthorityArn  --query CertificateArn --output text --profile backend) 
+export frontend_certificate_arn=$(aws acm request-certificate --domain-name "*.example.com" --certificate-authority-arn $certificateAuthorityArn  --query CertificateArn --output text --profile frontend)
+
+
+export frontendworkerrole=$(eksctl get iamidentitymapping --profile frontend --cluster am-multi-account-1  -o json |jq -r '.[].rolearn |split("/")|.[1]')
+export backendworkerrole=$(eksctl get iamidentitymapping --profile backend --cluster am-multi-account-2  -o json |jq -r '.[].rolearn |split("/")|.[1]')
+
+aws iam attach-role-policy --policy-arn arn:aws:iam::aws:policy/AWSCertificateManagerReadOnly --role-name $frontendworkerrole --profile frontend
+aws iam attach-role-policy --policy-arn arn:aws:iam::aws:policy/AWSCertificateManagerPrivateCAReadOnly --role-name $frontendworkerrole --profile frontend
+aws iam attach-role-policy --policy-arn arn:aws:iam::aws:policy/AWSAppMeshPreviewEnvoyAccess --role-name $frontendworkerrole --profile frontend
+
+
+aws iam attach-role-policy --policy-arn arn:aws:iam::aws:policy/AWSCertificateManagerReadOnly --role-name $backendworkerrole  --profile backend
+aws iam attach-role-policy --policy-arn arn:aws:iam::aws:policy/AWSCertificateManagerPrivateCAReadOnly --role-name $backendworkerrole --profile backend
+aws iam attach-role-policy --policy-arn arn:aws:iam::aws:policy/AWSAppMeshPreviewEnvoyAccess --role-name $backendworkerrole --profile backend
+
+
+
+sed -e  "s%CERTIFICATE_ARN%"$backend_certificate_arn"%g" -e "s%CA_ARN%"$certificateAuthorityArn"%g" -e "s%FRONTEND_ACCOUNT%"$frontendaccount"%g" redis_update > redis_update.json
+sed -e  "s%CERTIFICATE_ARN%"$backend_certificate_arn"%g" -e "s%CA_ARN%"$certificateAuthorityArn"%g" -e "s%FRONTEND_ACCOUNT%"$frontendaccount"%g" appserver_update > appserver_update.json
+sed -e  "s%CERTIFICATE_ARN%"$backend_certificate_arn"%g" -e "s%CA_ARN%"$certificateAuthorityArn"%g" -e "s%FRONTEND_ACCOUNT%"$frontendaccount"%g" yelb_db_update > yelb_db_update.json
+sed -e  "s%FRONTEND_CERTIFICATE_ARN%"$frontend_certificate_arn"%g" -e "s%CA_ARN%"$certificateAuthorityArn"%g" -e "s%FRONTEND_ACCOUNT%"$frontendaccount"%g" yelb_ui_update > yelb_ui_update.json
+
+
+aws appmesh update-virtual-node --cli-input-json file://redis_update.json --profile backend
+aws appmesh update-virtual-node --cli-input-json file://appserver_update.json --profile backend
+aws appmesh update-virtual-node --cli-input-json file://yelb_db_update.json --profile backend
+aws appmesh update-virtual-node --cli-input-json file://yelb_ui_update.json --profile frontend
```

Comment:
Why are we updating app mesh configuration outside of the CRDs? The controller is not going to be able to reconcile it since its knowledge of the world is different than what's mastered in the server.

These should be done via CRD updates I believe, which would get rid of the json update files in this PR.

Reply:
This is applying the TLS encryption, do we need custom resource definition for this ?

Reply:
As in, you should be updating the `kind: VirtualNode` CRDs via the manifests, maybe applying a new file. If resources are managed via the controller, they should _not_ be updated via the API/console.

I'm actually surprised this works as-is. The controller would reconcile the resources back to what it thinks they should be.

Reply:
Understood, do you have a sample reference for me to look at how to implement the virtual node updates via CRDs ?